### ISSUE TITLE
Problema com o Trabalho 2

### ISSUE THREAD

**Issue by goncalocesar01** [miguelriemoliveira/savi_22-23#15]
Olá, Boa noite
Tamos com problemas pois temos este erro "Erro: Unable to find a valid cuDNN algorithm to run convolution",
já tentamos alteramos o Batch size e image size mas ele continua.
Alguém sabe como podemos resolver isto.

[Ex1.zip](https://github.com/miguelriemoliveira/savi_22-23/files/10363058/Ex1.zip)

**Comment by JasCosta** [comment]
olá professor 
quando iniciamos o trainning  aparece o erro acima mencinado 
até ao treino o programa decorre bem

extrato do trainning
**# #     # -----------------------------------------------------------------
# #     # Training
# #     # -----------------------------------------------------------------
    
    loss_visualizer = DataVisualizer('Loss')
    loss_visualizer.draw([0,maximum_num_epochs], [termination_loss_threshold, termination_loss_threshold], layer='threshold', marker='--', markersize=1, color=[0.5,0.5,0.5], alpha=1, label='threshold', x_label='Epochs', y_label='Loss')

    test_visualizer = ClassificationVisualizer('Test Images')

    # Resume training
    if resume_training:
        checkpoint = torch.load(model_path)
        model.load_state_dict(checkpoint['model_state_dict'])
        optimizer.load_state_dict(checkpoint['optimizer_state_dict'])
        idx_epoch = checkpoint['epoch']
        epoch_train_losses = checkpoint['train_losses']
        epoch_test_losses = checkpoint['test_losses']

        model.train()    
if __name__ == "__main__":
    main()****

**Comment by miguelriemoliveira** [comment]
Olá @JasCosta ,

obrigado. Vou procurar.

**Comment by miguelriemoliveira** [comment]
Segundo isto 

https://stackoverflow.com/questions/61467751/unable-to-find-a-valid-cudnn-algorithm-to-run-convolution

o erro pode ser porque a memória máxima é atingida, e recomenda-se baixar o batch size. Já tentaram fazer isso?

**Comment by JasCosta** [comment]
bom dia 
professor com 800 imagens no dataset o erro mantem-se. 
tambem testei apenas com tipo de imagem no meu caso maça , e o erro manteve-se

quando teste apenas com maça apareceu tambem a msg 
**No artists with labels found to put in legend.  Note that artists whose label start with an underscore are ignored when legend() is called with no argument.**

**Comment by GSoares3925** [comment]
Bom dia,
Para mim com um dataset bastante mais pequeno (10 imagens )funciona até à linha 233 do main. Ainda não encontrei o limite de imagens que posso ter no dataset mas penso que se pode confirmar que se trata de um problema de memória. A unica mudança que fiz foi o valor de k nas linhas que utilizavam ramdom.sample para ser mais baixo que o numero de imagens no dataset.

O erro que me dá agora é:
Exception has occurred: RuntimeError
Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
  File "/home/gaspar/Desktop/Ex1/main.py", line 233, in main
    optimizer.step() # update the weights
  File "/home/gaspar/Desktop/Ex1/main.py", line 294, in <module>
    main()
    
    Entretanto vou tentar resolver e se conseguir volto a postar no issue.

**Comment by GSoares3925** [comment]
Alterando o device que utilizo na linha 41 resolveu o meu problema que utilizo (linha 41)
linha 41 antiga:
device = 'cuda: 0' if torch.cuda.is_available() else 'cpu' 
linha 41 nova:
device = 'cpu' if torch.cuda.is_available() else 'cuda: 0'

**Comment by miguelriemoliveira** [comment]
@JasCosta , você e o @GSoares3925 estão a trabalhar juntos? É o mesmo problema ou outro diferente?

**Comment by JasCosta** [comment]
Sim estamos a trabalhar juntos




cumprimentos,


José Costa

________________________________
De: Miguel Riem de Oliveira ***@***.***>
Enviado: 11 de janeiro de 2023 09:30
Para: miguelriemoliveira/savi_22-23 ***@***.***>
Cc: Jose Costa ***@***.***>; Mention ***@***.***>
Assunto: Re: [miguelriemoliveira/savi_22-23] Problema com o Trabalho 2 (Issue #15)


@JasCosta<https://github.com/JasCosta> , você e o @GSoares3925<https://github.com/GSoares3925> estão a trabalhar juntos? É o mesmo problema ou outro diferente?

**Comment by goncalocesar01** [comment]
ja consegui resolver o problema, o código esta com problema com um diretório e estava a deixar o target de torch.nn.CrossEntropyLoss() para foras dos limites dando problema de memoria
